provision/acc_provision/acc_provision.py

```diff
@@ -1026,6 +1026,37 @@ def config_validate_preexisting(config, prov_apic):
     return True
 
 
+def calico_config_validate_preexisting(config, prov_apic):
+    try:
+        if prov_apic is not None:
+            apic = get_apic(config)
+            if apic is None:
+                return False
+        vrf_tenant = config["aci_config"]["vrf"]["tenant"]
+        vrf_name = config["aci_config"]["vrf"]["name"]
+        vrf_dn = config["aci_config"]["vrf"]["dn"]
+        l3out_name = config["aci_config"]["l3out"]["name"]
+        tenant = apic.get_tenant(vrf_tenant)
+        if tenant is None:
+            err("Tenant %s not created on the APIC. Please create the tenant and try again" % (vrf_tenant))
+            return False
+        vrf = apic.get_vrf(vrf_dn)
+        if vrf is None:
+            err("VRF %s/%s not created on the APIC. Please create the vrf and try again" % (vrf_tenant, vrf_name))
+            return False
+        l3out = apic.get_l3out(vrf_tenant, l3out_name)
+        if l3out is None:
+            err("L3out %s/%s not created on the APIC. Please create the l3out and try again " % (vrf_tenant, l3out_name))
+            return False
+        map_l3out_vrf = apic.check_l3out_vrf(vrf_tenant, l3out_name, vrf_name, vrf_dn)
+        if not map_l3out_vrf:
+            err("VRF is not mapped to L3out %s/%s on the APIC. Please fix the configuration and try again" % (vrf_tenant, l3out_name))
+            return False
+    except Exception as e:
```

Comment:
need aep and phys dom checks as well..

Reply:
Thanks for pointing out, have updated it